### PR TITLE
docs: better document differences between deployment methods

### LINKED ISSUE #651: docs: document the purpose of different deployment options
**What would you like to be added**:

Better document the purpose of different deployment options.

Spun off of https://github.com/kubernetes-sigs/node-feature-discovery/pull/640#issuecomment-968687743

> I mean that we should probably accept the different deployment options for what they are. For example:
> - kustomize as the simplest "non-packaged" deployment method, aligned with the "deploy once" principle of the project, with sane defaults, but manual configuration and upgrades.
> - OLM operator (maybe) as the preferred way to deploy NFD cluster-scoped resources. Configuration is done using NFD CRDs, that other operators may add as a dependency (which is a very important feature).
> - Helm chart as a nice declarative way to deploy/revise NFD on any cluster, with easy configuration management and upgrades, but (currently) not ideal for other Helm apps to add as a dependency (since we may end up with the same NFD functionality deployed more than once in the same cluster).

**Why is this needed**:

Helpl users better understand which deployment option to pick. Help developers to better understand the deployment strategies.

### PR DESCRIPTION
**Pull request by marquiz**
Fixes #651

### CODE REVIEW

**Comment by marquiz** [comment]
Some improvement, I hope. Not necessarily perfect or complete.

/assign @ArangoGutierrez @fmuyassarov
ping @eliaskoromilas

File: docs/deployment/helm.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ sort: 3
 
 Node Feature Discovery Helm chart allow to easily deploy and manage NFD.
 
+> NOTE: NFD is not ideal for other Helm charts to depend on as that may result
+> in multiple parallel NFD deployments in the same cluster which is not fully
+> supported bu the NFD Helm chart.
```
Comment (eliaskoromilas) [inline docs/deployment/helm.md:22]:
```suggestion
> supported by the NFD Helm chart.
```

File: docs/deployment/index.md [hunk 2]
```diff
@@ -6,4 +6,14 @@ sort: 2
 
 # Deployment
 
-Deployment instructions.
+See [Image variants](image-variants) for description of the different NFD
+container images available.
+
+[Using Kustomize](kustomize) provides straightforward deployment with
+`kubectl` ntegration and declarative customization.
```
Comment (eliaskoromilas) [inline docs/deployment/index.md:13]:
```suggestion
`kubectl` integration and declarative customization.
```

File: docs/deployment/index.md [hunk 3]
```diff
@@ -6,4 +6,14 @@ sort: 2
...
+
+[Using Helm](helm) provides easy management of NFD deployments with nice
+onfiguration management and easy upgrades.
```
Comment (eliaskoromilas) [inline docs/deployment/index.md:16]:
```suggestion
configuration management and easy upgrades.
```

File: docs/deployment/helm.md [hunk 4]
```diff
@@ -17,6 +17,10 @@ sort: 3
...
+
```
Comment (fmuyassarov) [inline docs/deployment/helm.md:23]:
```suggestion
> supported by the NFD Helm chart.

```

**Review by eliaskoromilas — CHANGES_REQUESTED** [review]
Well-written!

File: docs/deployment/index.md (hunk 3 above)
Comment (fmuyassarov) [inline docs/deployment/index.md:16]:
```suggestion
configuration management and easy upgrades.
```

File: docs/deployment/index.md (hunk 2 above)
Comment (marquiz) [reply docs/deployment/index.md]: Thx, fixed!

File: docs/deployment/helm.md (hunk 4 above)
Comment (marquiz) [reply docs/deployment/helm.md]: Fixed

File: docs/deployment/index.md (hunk 3 above)
Comment (marquiz) [reply docs/deployment/index.md]: Fixed

Comment (marquiz) [reply docs/deployment/index.md]: Thx, fixed

File: docs/deployment/helm.md (hunk 1 above)
Comment (marquiz) [reply docs/deployment/helm.md]: Thx, fixed

**Comment by marquiz** [comment]
> Well-written

...and hastily as you saw from all the typos 🙄 

Thanks @eliaskoromilas and @fmuyassarov for really fast feedback. Comments addressed, PTAL

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**